packages/base/src/mainview/TemporalSlider.tsx

```diff
@@ -0,0 +1,430 @@
+
+  const layerIdRef = useRef('');
+  const intervalRef = useRef<NodeJS.Timeout | null>(null);
+
+  const { featureProperties } = useGetProperties({ layerId, model });
+
+  useEffect(() => {
+    const handleClientStateChanged = () => {
+      if (!model.localState?.selected?.value) {
+        return;
+      }
+
+      const selectedLayerId = Object.keys(model.localState.selected.value)[0];
+
+      // reset
+      if (selectedLayerId !== layerIdRef.current) {
+        setSelectedFeature('');
+        setRange({ start: 0, end: 1 });
+        setMinMax({ min: 0, max: 1 });
+        setValidFeatures([]);
+        setInferredDateFormat('yyyy-MM-dd');
+        setStep(stepMap.year);
+        setCurrentValue(0);
+        setFps(1);
+        setValidSteps({});
+        setLayerId(selectedLayerId);
+      }
+    };
+
+    const handleLayerChange = (
+      _: IJupyterGISDoc,
+      change: IJGISLayerDocChange
+    ) => {
+      // Get the changes for the selected layer
+      const selectedLayer = change.layerChange?.find(
+        layer => layer.id === layerIdRef.current
+      );
+
+      // Bail if there's no relevant change
+      if (!selectedLayer?.newValue) {
+        return;
+      }
+
+      const { newValue, oldValue } = selectedLayer;
+
+      // If layer was deleted (empty object) or the layer type changed, close the temporal controller
+      if (
+        Object.keys(newValue).length === 0 ||
+        newValue.type !== oldValue.type
+      ) {
+        model.toggleTemporalController();
+      }
+    };
+
+    // Initial state
+    handleClientStateChanged();
+
+    model.clientStateChanged.connect(handleClientStateChanged);
+    model.sharedLayersChanged.connect(handleLayerChange);
+
+    return () => {
+      model.clientStateChanged.disconnect(handleClientStateChanged);
+      model.sharedLayersChanged.disconnect(handleLayerChange);
+      removeFilter();
+      if (intervalRef.current) {
+        clearInterval(intervalRef.current);
+      }
+    };
+  }, []);
+
+  useEffect(() => {
+    layerIdRef.current = layerId;
+  }, [layerId]);
+
+  useEffect(() => {
+    const featuresForSelect = [];
+
+    // We only want to show features that could be time values
+    for (const [key, set] of Object.entries(featureProperties)) {
+      let checkIfDateIsValid = false;
+      const checkValue = set.values().next().value;
+
+      // We only want to look at strings and whole numbers
+      // ? Is there a better way to check if number values are valid timestamps?
+      // ! QGIS doesn't actually support number values for their time thing
+      const isString = typeof checkValue === 'string';
+      const isNumber =
+        typeof checkValue === 'number' && Number.isInteger(checkValue);
+      if (!isString && !isNumber) {
+        continue;
+      }
+
+      checkIfDateIsValid = isValid(toDate(checkValue));
+      if (!checkIfDateIsValid) {
+        continue;
+      }
+
+      if (checkValue && checkIfDateIsValid) {
```

Comment:
```suggestion
      if (checkValue) {
```

We know `checkIfDateIsValid` is truthy at this point.